### DISCUSSION TITLE
not working with graphql-sse

### DISCUSSION THREAD

**Discussion by sharmapukar217** [urql-graphql/urql#2697] [Q&A]
data returned from useSubscription is undefined  when using graphql-sse for subscription over server sent events

**Comment by JoviDeCroock** [comment]
This is a bit low-effort isn't it 😅 the general discussions aren't really meant for bug reports, you could leverage issues or q&a but as this looks like a bug a reproduction would be very welcomed.

**Reply by sharmapukar217** [reply]
But I don't think this is a bug. I might be missing something so created a discussion

**Reply by JoviDeCroock** [reply]
Hard to help without an example, ... but from the [`graphql-sse`](https://github.com/enisdenjo/graphql-sse#urql) repository I can see an urql example

```js
import { createClient, defaultExchanges, subscriptionExchange } from 'urql';
import { createClient as createWSClient } from 'graphql-sse';

const sseClient = createWSClient({
  url: 'http://its.urql:4000/graphql/stream',
});

const client = createClient({
  url: '/graphql/stream',
  exchanges: [
    ...defaultExchanges,
    subscriptionExchange({
      forwardSubscription(operation) {
        return {
          subscribe: (sink) => {
            const dispose = sseClient.subscribe(operation, sink);
            return {
              unsubscribe: dispose,
            };
          },
        };
      },
    }),
  ],
});
```
